RoddyCore/src/de/dkfz/roddy/execution/io/fs/BashCommandSet.groovy

```diff
@@ -122,33 +127,79 @@ class BashCommandSet extends ShellCommandSet {
         else
+            // I do not understand, why this command ignores execution errors (via `|| echo ''`).
             return "${checkExistence} || install -d \"${path}\" || echo ''"
     }
 
-    @Override
-    String getCheckChangeOfPermissionsPossibilityCommand(File f, String group) {
-        File testFile = new File(f, ".roddyPermissionsTestFile")
-        return "(touch ${testFile}; chmod u+rw ${testFile} &> /dev/null && chgrp ${group} ${testFile} &> /dev/null) $TRUE_OR_FALSE; rm ${testFile} 2>/dev/null; echo ''"
+    /** Check whether it is possible to change permissions at the target-location. Specifically, the path `file` will
+     *  be appended by ".roddyPermissionTestFile" and it is checked, whether this can get changes of the permissions.
+     *  Currently, it is just checked whether chmod is working on that file and -- if group is not null -- whether
+     *  the group ownership can be changed to the target group.
+     *
+     * @param file    path to check. Not exactly this path is changed, but a file besides that path.
+     * @param group   target-group to check; if this is null, then only chmod is tested, otherwise also chgrp.
+     * @return
+     */
+    @Override
+    String getCheckChangeOfPermissionsPossibilityCommand(File file, String group) {
+        File testFile = new File(file, ".roddyPermissionsTestFile")
+        if (null == group) {
+            return "(touch ${testFile}; chmod u+rw ${testFile} &> /dev/null) $TRUE_OR_FALSE; rm ${testFile} 2>/dev/null; echo ''"
+        } else {
+            return "(touch ${testFile}; chmod u+rw ${testFile} &> /dev/null && chgrp ${group} ${testFile} &> /dev/null) $TRUE_OR_FALSE; rm ${testFile} 2>/dev/null; echo ''"
+        }
     }
 
     @Override
-    String getSetAccessRightsCommand(File f, String rightsForFiles, String fileGroup) {
-        return "chmod ${rightsForFiles} ${f.absolutePath}; chgrp ${fileGroup} ${f.absolutePath}"
+    Optional<String> getSetAccessRightsCommand(File f, String rightsForFiles, String fileGroup) {
+        def path = "${f.absolutePath}"
+        String result = ""
+        if (null != rightsForFiles)
+            result += "chmod ${rightsForFiles} ${path}; "
+        if (null != fileGroup)
+            result += "chgrp ${fileGroup} ${path}"
+        return Optional.ofNullable(result == "" ? null : result)
     }
 
     @Override
-    String getSetAccessRightsRecursivelyCommand(File f, String rightsForDirectories, String rightsForFiles, String fileGroup) {
-        def path = "${f.getAbsolutePath()}"
-        return "find ${path} -type d | xargs chmod ${rightsForDirectories}; find ${path} -type d | xargs chgrp ${fileGroup}; find ${path} -type f | xargs chmod ${rightsForFiles}; find ${path} -type f | xargs chgrp ${fileGroup};"
+    Optional<String> getSetAccessRightsRecursivelyCommand(File f, String rightsForDirectories, String rightsForFiles, String fileGroup) {
```

Comment:
This here (and below) fixes the problem with `null` parameter values provided to these methods in the old Roddy code and failing calls like `chgrp null $file` that resulted from it.